### PR TITLE
Improve nested and empty proc support

### LINKED ISSUE #16: Support nested blocks
I picked up the hello world example form the Ruby docs which appears to fail with prettier-ruby:

https://apidock.com/rails/ActionController/MimeResponds/InstanceMethods/respond_to

```ruby
def index
  @people = Person.find(:all)

  respond_to do |format|
    format.html
    format.xml { render :xml => @people.to_xml }
  end
end
```

Seems to crash prettier:

```
~/prettier-ruby/vendor/ruby/astexport.rb:29:in `visit': undefined method `first' for nil:NilClass (NoMethodError)
  from ~/prettier-ruby/vendor/ruby/astexport.rb:368:in `visit_brace_block'
  from ~/prettier-ruby/vendor/ruby/astexport.rb:139:in `visit'
  from ~/prettier-ruby/vendor/ruby/astexport.rb:131:in `visit'
  from ~/prettier-ruby/vendor/ruby/astexport.rb:491:in `block in visit_exps'
  from ~/prettier-ruby/vendor/ruby/astexport.rb:489:in `each'
  from ~/prettier-ruby/vendor/ruby/astexport.rb:489:in `visit_exps'
  from ~/prettier-ruby/vendor/ruby/astexport.rb:137:in `visit'
  from ~/prettier-ruby/vendor/ruby/astexport.rb:131:in `visit'
  from ~/prettier-ruby/vendor/ruby/astexport.rb:491:in `block in visit_exps'
  from ~/prettier-ruby/vendor/ruby/astexport.rb:489:in `each'
  from ~/prettier-ruby/vendor/ruby/astexport.rb:489:in `visit_exps'
  from ~/prettier-ruby/vendor/ruby/astexport.rb:480:in `visit_bodystmt'
  from ~/prettier-ruby/vendor/ruby/astexport.rb:190:in `visit'
  from ~/prettier-ruby/vendor/ruby/astexport.rb:440:in `visit_def'
  from ~/prettier-ruby/vendor/ruby/astexport.rb:156:in `visit'
  from ~/prettier-ruby/vendor/ruby/astexport.rb:491:in `block in visit_exps'
  from ~/prettier-ruby/vendor/ruby/astexport.rb:489:in `each'
  from ~/prettier-ruby/vendor/ruby/astexport.rb:489:in `visit_exps'
  from ~/prettier-ruby/vendor/ruby/astexport.rb:31:in `visit'
  from ~/prettier-ruby/vendor/ruby/astexport.rb:11:in `initialize'
  from ~/prettier-ruby/vendor/ruby/astexport.rb:747:in `new'
  from ~/prettier-ruby/vendor/ruby/astexport.rb:747:in `<main>'

### PR DESCRIPTION
**Pull request by AlanFoster**
Add more tests and fix some edge cases with missing params on procs

Closes https://github.com/iamsolankiamit/prettier-ruby/issues/16

### CODE REVIEW

File: tests/block/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -1,16 +1,5 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[`block_1.rb 1`] = `
```
Comment (AlanFoster) [inline tests/block/__snapshots__/jsfmt.spec.js.snap:3]: This was moved down for some reason, but the output is the same 👌

**Comment by AlanFoster** [comment]
Closing this until the Jest upgrade is in, as it creates a different snapshot file format
